Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/PatientSearchBean.java

```diff
@@ -0,0 +1,606 @@
+ */
+@ManagedBean
+@SessionScoped
+public class PatientSearchBean {
+
+    private static final Logger LOG = Logger.getLogger(PatientSearchBean.class);
+
+    //Generic Variables, can be moved to a constant file
+    private static final String PATIENT_FOUND = "Patient Record Found:";
+    private static final String PATIENT_NOT_FOUND = "Patient Not Found";
+    private static final String DOCUMENT_NOT_FOUND = "No Documents Found";
+    private static final String DOCUMENT_FOUND = "Documents Found";
+    private static final String PATIENT_SEARCH_LOAD_MESSAGE = "Searching Patient....";
+    private static final String DOCUMENT_SEARCH_LOAD_MESSAGE = "Searching Patient Documents....";
+    private static final String DISPLAYING_LOAD_MESSAGE = "Retrieving Patient Document....";
+    private static final String CONTENT_TYPE_IMAGE_PNG = "image/png";
+    private static final String CONTENT_TYPE_IMAGE_JPEG = "image/jpeg";
+    private static final String CONTENT_TYPE_IMAGE_GIF = "image/gif";
+    private static final String CONTENT_TYPE_TEXT_XML = "text/xml";
+    private static final String CONTENT_TYPE_TEXT_PLAIN = "text/plain";
+    private static final String CONTENT_TYPE_TEXT_HTML = "text/html";
+    private static final String CONTENT_TYPE_APPLICATION_XML = "application/xml";
+    private static final String CONTENT_TYPE_APPLICATION_XHTML_XML = "application/xhtml+xml";
+    private static final String CONTENT_TYPE_APPLICATION_OCTET_STREAM = "application/octet-stream";
+    private static final String CONTENT_TYPE_APPLICATION_PDF = "application/pdf";
+
+    private int activeIndex = 0;
+    private Document selectedCurrentDocument;
+
+    private StreamedContent documentImage;
+    private StreamedContent documentPdf;
+    private boolean renderDocumentimage;
+    private boolean renderDocumentPdf;
+    private boolean renderDocumentText;
+    private boolean renderDcoumentNotSupported;
+
+    //For Lookup..should be moved to a different managed bean
+    private List<SelectItem> documentTypeList;
+    private Map<String, String> organizationList;
+    private Map<String, String> genderList;
+
+    //Used in Patient Query Page
+    private boolean patientFound;
+    private String patientMessage;
+    private String firstName;
+    private String lastName;
+    private Date dateOfBirth;
+    private String gender;
+    private String organization;
+    private List<Patient> patientList;
+    private int selectedPatient;
+
+    //Used in Document Query Page
+    private boolean documentFound;
+    private Date documentRangeFrom;
+    private Date documentRangeTo;
+    private String documentMessage;
+    private List<String> querySelectedDocuments;
+    private int selectedDocument;
+
+    //TODO: Temporary should be removed, should use the patient object documentList
+    private List<Document> documentList;
+
+    /**
+     * Instantiate all the variables and load the lookup Data
+     */
+    public PatientSearchBean() {
```

Comment:
should we consider using @postconstruct instead of having all of this loaded in the constructor?